Add documentation for Exemplars (#30317)
grafana/grafana: docs/sources/datasources/prometheus.md

| `Scrape interval`         | Set this to the typical scrape and evaluation interval configured in Prometheus. Defaults to 15s.                                                                                                       |
| `Disable metrics lookup`  | Checking this option will disable the metrics chooser and metric/label support in the query field's autocomplete. This helps if you have performance issues with bigger Prometheus instances.           |
| `Custom Query Parameters` | Add custom parameters to the Prometheus query URL. For example `timeout`, `partial_response`, `dedup`, or `max_source_resolution`. Multiple parameters should be concatenated together with an '&amp;'. |

> | `Label name`              | The name of the field in the labels object that should be used.                                                                                                                                         |

Reviewer: ```suggestion
| `Label name`              | Add the name of the field in the label object.                                                                                                                                         |
```

| `URL`                     | If the link is external, then enter the full link URL. You can interpolate the value from the field with `${__value.raw }` macro.                                                                       |

> | `Internal link`           | Select if the link is internal or external. In case of internal link, a data source selector allows you to select the target data source. Only tracing data sources are supported.                      |

Reviewer: ```suggestion
| `Internal link`           | Select if the link is internal or external. In the case of an internal link, a data source selector allows you to select the target data source. Only supports tracing data sources.                      |
```

| `Disable metrics lookup`  | Checking this option will disable the metrics chooser and metric/label support in the query field's autocomplete. This helps if you have performance issues with bigger Prometheus instances.           |
| `Custom Query Parameters` | Add custom parameters to the Prometheus query URL. For example `timeout`, `partial_response`, `dedup`, or `max_source_resolution`. Multiple parameters should be concatenated together with an '&amp;'. |

### Exemplars

> 

Reviewer 2: ```suggestion

> **Note:** This feature is available in Grafana 7.4+.
```

Reviewer 2: Or do you mean that it is not in the current Prometheus version?

Author: It is not in Prometheus yes.

Author: I've added a note with the Prometheus version should we add one with Grafana as well?

Reviewer 2: Yes

| `Disable metrics lookup`  | Checking this option will disable the metrics chooser and metric/label support in the query field's autocomplete. This helps if you have performance issues with bigger Prometheus instances.           |
| `Custom Query Parameters` | Add custom parameters to the Prometheus query URL. For example `timeout`, `partial_response`, `dedup`, or `max_source_resolution`. Multiple parameters should be concatenated together with an '&amp;'. |

### Exemplars

> **Note:** This feature is available in Prometheus 2.25+.

> The Exemplars configuration allows you to add multiple endpoints to any of your Exemplar label.

Reviewer: This looks a little bit out of place. Perhaps we can introduce exemplars after the introduction. Then add them into the various sections.

You can use this functionality to link to your tracing backend from an Exemplar.

> {{< docs-imagebox img="/img/docs/v74/exemplars-settings.png" class="docs-image--no-shadow" caption="Screenshot of the Exemplars configuration" >}}

Reviewer: It would be great if we can add a link to tracing doc content from here.

Author: Which one?

Reviewer: You can list the data sources and explore where the "trace to logs" feature is available. 

https://grafana.com/docs/grafana/latest/datasources/tempo/#trace-to-logs
https://grafana.com/docs/grafana/latest/explore/#trace-to-logs
https://grafana.com/docs/grafana/latest/datasources/zipkin/#trace-to-logs
https://grafana.com/docs/grafana/latest/datasources/jaeger/#trace-to-logs

Author: I'm not sure how the trace-to-logs feature is related to this sentence. Can you help me rephrase it?

Each Exemplar link config consists of:

- **Label name -** The name of the field in the labels object that should be used.
- **URL -** If the link is external, then enter the full link URL. You can interpolate the value from the field with `${__value.raw }` macro.

> - **Internal link -** Select if the link is internal or external. In case of internal link, a data source selector allows you to select the target data source. Only tracing data sources are supported.

Reviewer: This should be in the table above for add data source configuration.

Author: I didn't add this to the configuration table to be able to distinguish these configs as these are for exemplars. I've seen that we did this in Loki documentation for derived links. What do you think?

Reviewer: @zoltanbedi, when I installed the beta build, and I was adding Prometheus data source, I saw these options. They should be in the table, with a reference to Exemplars content. The content itself should follow the introduction section. I do not think it makes sense to keep the content separate. It can make the content choppy.

| `Query expression`  | Prometheus query expression, check out the [Prometheus documentation](http://prometheus.io/docs/querying/basics/).                                                                                                                                                                                                                                                                                                                                                                                        |
| `Legend format`     | Controls the name of the time series, using name or pattern. For example `{{hostname}}` is replaced with the label value for the label `hostname`.                                                                                                                                                                                                                                                                                                                                                        |

> | `Min step`          | An additional lower limit for the [`step` parameter of Prometheus range queries](https://prometheus.io/docs/prometheus/latest/querying/api/#range-queries) and for the `$__interval` and `$__rate_interval` variables. The limit is absolute and not modified by the _Resolution_ setting.                                                                                                                                                                                                                |

Reviewer: ```suggestion
| `Min step`          | An additional lower limit for the [`step` parameter of Prometheus range queries](https://prometheus.io/docs/prometheus/latest/querying/api/#range-queries) and for the `$__interval` and `$__rate_interval` variables. The limit is absolute, it cannot modified by the _Resolution_ setting.                                                                                                                                                                                                                |
```

```

## Amazon Managed Service for Prometheus

The Prometheus data source works with Amazon Managed Service for Prometheus. If you are using an AWS Identity and Access Management (IAM) policy to control access to your Amazon Managed Service for Prometheus domain, then you must use AWS Signature Version 4 (AWS SigV4) to sign all requests to that domain. For more details on AWS SigV4, refer to the [AWS documentation](https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html).

> **Note:** Grafana version 7.3.5 or higher is required to use SigV4 authentication.

To connect the Prometheus data source to Amazon Managed Service for Prometheus using SigV4 authentication, refer to the AWS guide to [Set up Grafana open source or Grafana Enterprise for use with AMP](https://docs.aws.amazon.com/prometheus/latest/userguide/AMP-onboard-query-standalone-grafana.html).

If you are running Grafana in an Amazon EKS cluster, follow the AWS guide to [Query using Grafana running in an Amazon EKS cluster](https://docs.aws.amazon.com/prometheus/latest/userguide/AMP-onboard-query-grafana-7.3.html).

## Exemplars

> **Note:** This feature is available in Prometheus 2.25+ and Grafana 7.4+.

> You can use this functionality to link to your tracing backend from an Exemplar.

Reviewer: Can we add more information here? This is a little sparse. Thanks!
